submissions/madmaxWMFU/js-movie-booking/app.js

```diff
@@ -0,0 +1,156 @@
+const deleteFromCart = (id) => {
+    Array.from(cart.children).forEach((item) => {
+        if(item.getAttribute('id') === id) {
+           item.remove(); 
+        }
+    })
+}
+
+const changeSeatStatus = (seat, type) => seat.nextElementSibling.classList.toggle(`${type}--active`);
+
+const addToCart = ({target}) => {
+    if(target.classList.contains('hall__seat-data')) {
+        const data = target.dataset.value;
+        const [rowTicket, seatTicket, typeTicket] = data.split('-');
+        isCartEmpty()
+
+        if(target.checked) {
+            changeSeatStatus(target, typeTicket);
+            renderCart(data);
+            addSelectedTicket(rowTicket, seatTicket, typeTicket);
+            setTotalPrice();
+        } else {
+            changeSeatStatus(target, typeTicket); 
+            deleteFromCart(data);
+            removeSelectedTicket(rowTicket, seatTicket, typeTicket);
+            setTotalPrice();
+        }
+    }
+} 
+
+const addSelectedTicket = (row, seat, type) => {
+    let data = JSON.parse(window.sessionStorage.getItem('serverData'));
+    const ticket = {
+        "row": row,
+        "seat": seat,
+        "type": type
+    }
+    data.selected.push(ticket);
+    window.sessionStorage.setItem('serverData', JSON.stringify(data));
+}
+
+const removeSelectedTicket = (row, seat) => {
+    let data = JSON.parse(window.sessionStorage.getItem('serverData'));
+    data.selected = data.selected.filter((ticket) => {
+        if(`${ticket.row}-${ticket.seat}` !== `${row}-${seat}`) {
+            return ticket;
+        }
+    })
+    window.sessionStorage.setItem('serverData', JSON.stringify(data));
+}
+
+const getSelectedTikets = () => JSON.parse(window.sessionStorage.getItem('serverData')).selected;
+
+const isCartEmpty = () => {
+    let selectedTikets = getSelectedTikets();
+
+    if(selectedTikets.length === 0) {
+        emptyCart.classList.toggle('no-display');
+        fullCart.classList.toggle('no-display');
+        return true;
+    }
+
+    return false;
+}
+
+const setTotalPrice = () => {
+    if(!isCartEmpty()) {
+        let selectedTikets = getSelectedTikets();
+        let countTikets = selectedTikets.length;
+        let initialSum= 0;
+        let totalPrice = selectedTikets.reduce((sum, price) => {
+            return sum + ticketPrice[price.type];
+        }, initialSum);
+
+        priceCart.innerText = `Кількість: ${countTikets} шт. Ціна: ${totalPrice} грн.`; 
+    }
+}
+
+const createJSON = () => {
+    const cart =  {
+        "film": filmTitle.innerText,
+        "datetime": `${reverseDate(filmDate.innerText)} ${filmTime.innerText}`,
+        "hall": filmHall.innerText,
+        "selected": []
+    }
+
+    window.sessionStorage.setItem("serverData", JSON.stringify(cart));
+}
+
+const init = async () => {
+    renderData(await getData());
+    createJSON();
+    hall.addEventListener('click', addToCart);
+    hall.addEventListener('touchstart', addToCart, {capture: true});
+}
+
+window.onload = init();
```

Comment:
Avoid adding event listeners using `onevent` attribute use addEventListener for this.

Reply:
Not fixed yet.